src/gui/settingsdialog.cpp

```diff
@@ -267,7 +265,20 @@ SettingsDialog::~SettingsDialog()
 
 QSize SettingsDialog::minimumSizeHint() const
 {
-    return {800, 500};
+    const auto screen = windowHandle() ? windowHandle()->screen() : QApplication::screenAt(QCursor::pos());
+    const auto geometry = screen->availableSize();
+    const QSize min { 800, 600 };
+    const QSize currentMin { static_cast<int>(geometry.width() * 0.75), static_cast<int>(geometry.height() * 0.75) };
+    if (currentMin.width() < min.width() || currentMin.height() < min.height()) {
```

Comment:
This line is confusing. Why use `currentMin` if it's smaller than `min`? Shouldn't it be the other way round, i.e., one should use the larger of both?